### PR TITLE
chore: output miner status

### PR DESCRIPTION
**Pull request by quake**
### What problem does this PR solve?

outputs miner status in cli cmd

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

```
uncle_rate: 153507/5920421(0.025928392592351118)
by_miner_script:
percentage | total | args                                     | code_hash                                                        | hash_type 
35.9%      | 359   | dde7801c073dfb3464c7b1f05b806bb2bbb84e99 | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
18.6%      | 186   | 7164f48d7a5bf2298166f8d81b81ea4e908e16ad | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
14.0%      | 140   | ba6caca0c7e893e412d67264c2eb2a1fc13c46fd | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
11.3%      | 113   | 4a336470564d07ca7059b7980481c2d59809d637 | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
8.2%       | 82    | 23ca27e4b95f962e4dd6618fb5fd79cf70006597 | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
5.2%       | 52    | 787725429829c280dbd225d8627fc88f57b0eaa5 | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
2.3%       | 23    | ed99d08d751ae32df7fc6f3048b7cca56c3bfafa | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
1.6%       | 16    | a528f2b9a51118b193178db4cf2f3db92e7df323 | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
1.2%       | 12    | 60cdc4aa67a4543f4246fba0cfcd55859c5d3ddd | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
0.9%       | 9     | 131d807770e13558a387ead6ddad83f77a99f4b8 | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
0.8%       | 8     | 87c713e60afa56ca8711942fc57e9666d7474694 | 9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 | Type
by_miner_message:
percentage | total | messages                                          
35.9%      | 359   | b"0.101.1 (6d73bd2-dirty 2021-10-29) \0\0\0\0"
30.7%      | 307   | b"0.101.1 (fb1e5dd 2021-10-27)"
19.2%      | 192   | b""
8.2%       | 82    | b"\x124Vx\x90"
2.3%       | 23    | b"0.101.1 (fb1e5dd 2021-10-27) MinedByViaBTC"
2.1%       | 21    | b"0.101.1 (fb1e5dd 2021-10-27) BixinKelePool"
1.6%       | 16    | b"\0\0\0\0"
```

to avoid introducing bench32 dependencies, this PR only output miner scripts in hex format instead of address string, user may use the ckb-cli to get the address:

```
ckb-cli util key-info --lock-arg dde7801c073dfb3464c7b1f05b806bb2bbb84e99
address:
  mainnet: ckb1qyqdmeuqrsrnm7e5vnrmruzmsp4m9wacf6vsxasryq

```

### Release note 

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

**Review by zhangsoledad — APPROVED** [review]

**Review by keroro520 — CHANGES_REQUESTED** [review]
Hexadecimal 'script' is less readable(maybe useless). I suggest using the expanded style: (lock.args, lock.code_hash, lock.hash_type)

File: ckb-bin/src/subcommand/stats.rs [hunk 1]
```diff
@@ -63,4 +66,57 @@ impl Statics {
...
+            by_miner_script
+                .entry(cellbase_witness.lock())
+                .and_modify(|e| *e += 1)
+                .or_insert(0);
+            by_miner_message
+                .entry(cellbase_witness.message().raw_data())
+                .and_modify(|e| *e += 1)
+                .or_insert(0);
+        }
+        let total = self.to - from;
```
Comment (chanhsu001) [inline ckb-bin/src/subcommand/stats.rs:95]: well, i now know why or_insert(0)....but i feel confused at first glance.

**Review by chanhsu001 — COMMENTED** [review]
Here i help to add helper function to resolve the script binary....i paste code here, to save time

```
use lazy_static::lazy_static;
lazy_static! {
    static ref WELL_KNOWN_SCRIPT: HashMap<(&'static str, &'static str), &'static str> = {
        let mut h = HashMap::new();
        h.insert(("Type", "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8"), "secp256k1-blake160-sighash-all");
        h.insert(("Data", "0x709f3fda12f561cfacf92273c57a98fede188a3f1a59b1f888d113f9cce08649"), "secp256k1-blake160-sighash-all");
        h.insert(("Type", "0x5c5069eb0857efc65e1bca0c07df34c31663b3622fd3876c876320fc9634e2a8"), "secp256k1-blake160-multisig-all");
        h.insert(("Data", "0x43400de165f0821abf63dcac299bbdf7fd73898675ee4ddb099b0a0d8db63bfb"), "secp256k1-blake160-multisig-all");
        h.insert(("Type", "0x82d76d1b75fe2fd9a27dfbaa65a039221a380d76c926f378d3f81cf3e7e13f2e"), "Dao");
        h.insert(("Data", "0x32064a14ce10d95d4b7343054cc19d73b25b16ae61a6c681011ca781a60c7923"), "Dao");
        h
    };
}
fn get_type(script: &Script) -> &str {
    if Into::<u8>::into(script.hash_type()) == Into::<u8>::into(ckb_types::core::ScriptHashType::Type) {
        "Type"
    } else if Into::<u8>::into(script.hash_type()) == Into::<u8>::into(ckb_types::core::ScriptHashType::Data) {
        "Data"
    } else if Into::<u8>::into(script.hash_type()) == Into::<u8>::into(ckb_types::core::ScriptHashType::Data1)
    {
        "Data1"
    } else {
        "none match"
    }
}

// check if match ckb well known script, output correspond script name, instead of binary data
// if not match, just return Display format string
fn to_readable_format(script: &Script) -> String {
    WELL_KNOWN_SCRIPT
        .get(&(get_type(script), format!("{:#x}", script.code_hash()).as_str()))
        .map_or_else(
            || format!("{}", script),
            |ckb_script_name| format!("{}, ScriptType:{}", ckb_script_name, get_type(script)),
        )
}
```

File: ckb-bin/src/subcommand/stats.rs [hunk 2]
```diff
@@ -63,4 +66,57 @@ impl Statics {
...
+
+        let mut by_miner_script_vec: Vec<_> = by_miner_script.into_iter().collect();
+        by_miner_script_vec.sort_by_key(|(_, v)| *v);
+        println!("by_miner_script:");
+        for (script, count) in by_miner_script_vec.iter().rev() {
+            println!(
+                "{}\t{}\t{:?}",
+                format!("{:.2}%", 100.0 * (*count as f64) / (total as f64)),
+                count,
+                script,
```
Comment (chanhsu001) [inline ckb-bin/src/subcommand/stats.rs:105]: more readable to  change 'script' to 'to_readable_format(script)'

**Comment by quake** [comment]
@keroro520 @chanhsu001 I have modified the code according to your review, please check the updated PR description and review again, thanks.

**Review by chanhsu001 — APPROVED** [review]
It goods to me.

**Review by keroro520 — APPROVED** [review]

**Comment by keroro520** [comment]
bors r=zhangsoledad,keroro520,chanhsu001

**Pull request merged**